code/backend/batch/utilities/integrated_vectorization/AzureSearchIndexer.py

```diff
@@ -42,9 +42,16 @@ def create_or_update_indexer(self, indexer_name: str, skillset_name: str):
         )
         return indexer_result
 
-    # To be updated for 'Reprocess All'
-    def run_indexer(self, indexer_name: str):
-        self.indexer_client.run_indexer(indexer_name)
-        logger.info(
-            f" {indexer_name} is created and running. If queries return no results, please wait a bit and try again."
-        )
+    def reprocess_all(self, indexer_name: str):
+        reprocess_response = {"status": "success"}
+        if indexer_name in [name for name in self.indexer_client.get_indexer_names()]:
+            self.indexer_client.reset_indexer(indexer_name)
+            run_response = self.indexer_client.run_indexer(indexer_name)
+            logger.info(
+                f" {indexer_name} is created and running. If queries return no results, please wait a bit and try again."
+            )
+            return reprocess_response if run_response is None else {"status": "error"}
```

Comment:
Isn't `run_response` always `None`?

Reply:
True, it is always None. I could not find any docs about it failing but because it is essentially an API call, I added the error condition in case anything goes wrong

Reply:
If we want to do error handling here we should do try/except rather than look at the return value